docs/code/MlNetCookBook.md

```diff
@@ -0,0 +1,212 @@
+});
+
+// Now read the file.
+var data = reader.Read(new MultiFileSource(dataPath));
+```
+
+## How do I load data with many columns from a CSV?
+`TextLoader` is used to load data from text files. You will need to specify what are the data columns, what are their types, and where to find them in the text file. 
+
+When the input file contains many columns of the same type, always intended to be used together, we recommend reading them as a *vector column* from the very start: this way the schema of the data is cleaner, and we don't incur unnecessary performance costs.
+
+Example file (https://github.com/dotnet/machinelearning/blob/master/test/data/generated_regression_dataset.csv):
+```
+-2.75,0.77,-0.61,0.14,1.39,0.38,-0.53,-0.50,-2.13,-0.39,0.46,140.66
+-0.61,-0.37,-0.12,0.55,-1.00,0.84,-0.02,1.30,-0.24,-0.50,-2.12,148.12
+-0.85,-0.91,1.81,0.02,-0.78,-1.41,-1.09,-0.65,0.90,-0.37,-0.22,402.20
+0.28,1.05,-0.24,0.30,-0.99,0.19,0.32,-0.95,-1.19,-0.63,0.75,443.51
+```
+
+Reading this file using `TextLoader`:
+```c#
+// Create a new environment for ML.NET operations. It can be used for exception tracking and logging, 
+// as well as the source of randomness.
+var env = new LocalEnvironment();
+
+// Create the reader: define the data columns and where to find them in the text file.
+var reader = TextLoader.CreateReader(env, ctx => (
+        // We read the first 10 values as a single float vector.
+        FeatureVector: ctx.LoadFloat(0, 9),
+        // Separately, read the target variable.
+        Target: ctx.LoadFloat(10)
+    ),
+    // Default separator is tab, but we need a comma.
+    separator: ',');
+
+
+// Now read the file.
+var data = reader.Read(new MultiFileSource(dataPath));
+```
+
+
+If the schema of the data is not known at compile time, or too cumbersome, you can revert to the dynamically-typed API: 
+```c#
+// Create a new environment for ML.NET operations. It can be used for exception tracking and logging, 
+// as well as the source of randomness.
+var env = new LocalEnvironment();
+
+// Create the reader: define the data columns and where to find them in the text file.
+var reader = new TextLoader(env, new TextLoader.Arguments
+{
+    Column = new[] {
+	    // We read the first 10 values as a single float vector.
+        new TextLoader.Column("FeatureVector", DataKind.R4, new[] {new TextLoader.Range(0, 9)}),
+        // Separately, read the target variable.
+        new TextLoader.Column("Target", DataKind.R4, 10)
+    },
+    // Default separator is tab, but we need a comma.
+    Separator = ","
+});
+
+// Now read the file. 
+var data = reader.Read(new MultiFileSource(dataPath));
+```
+
+## How do I look at the intermediate data?
+
+Oftentimes, when we construct the experiment, we want to make sure that the data processing 'up to a certain moment' produces the results that we want. With ML.NET it is not very easy to do: since all ML.NET operations are lazy, the objects we construct are just 'promises' of data.
```

Comment:
We should think about how to make that easier. Perhaps a simple wrapper for `head()` that does what the example below shows.

Reply:
we have some more in the works for it

---
In reply to: [219396398](https://github.com/dotnet/machinelearning/pull/960#discussion_r219396398) [](ancestors = 219396398)